qa/workunits/cephtool/test.sh

```diff
@@ -2726,6 +2726,14 @@ function test_osd_compact()
   $SUDO ceph daemon osd.1 compact
 }
 
+function test_osd_daemon_histogram()
+{
+  ceph -h
+  $SUDO ceph -h
+  ceph daemonhistogram osd.all osd.op_r_latency_out_bytes_histogram batch 1 3
```

Comment:
```
2021-08-16T20:03:44.594 INFO:tasks.workunit.client.0.smithi125.stdout:daemonhistogram {daemons} {name} [{mode} [{interval} [{count}]]]
2021-08-16T20:03:44.594 INFO:tasks.workunit.client.0.smithi125.stdout:                        Visualize Ceph daemons' performance histogram.
2021-08-16T20:03:44.594 INFO:tasks.workunit.client.0.smithi125.stdout:                        {daemons} specify where collect data from. This can be
2021-08-16T20:03:44.595 INFO:tasks.workunit.client.0.smithi125.stdout:                          either 'daemon_type.all' specification to use every
2021-08-16T20:03:44.595 INFO:tasks.workunit.client.0.smithi125.stdout:                          daemon of a specific type. Or a comma separated list
2021-08-16T20:03:44.595 INFO:tasks.workunit.client.0.smithi125.stdout:                          of daemon ids or admin socket paths.
2021-08-16T20:03:44.595 INFO:tasks.workunit.client.0.smithi125.stdout:                        {name} is a fully qualified perf histogram name,
2021-08-16T20:03:44.595 INFO:tasks.workunit.client.0.smithi125.stdout:                          full list of names is available through
2021-08-16T20:03:44.596 INFO:tasks.workunit.client.0.smithi125.stdout:                           'ceph daemon <daemon_name> perf histogram schema'
2021-08-16T20:03:44.596 INFO:tasks.workunit.client.0.smithi125.stdout:                        {mode} is either live(default) or batch output mode.
2021-08-16T20:03:44.596 INFO:tasks.workunit.client.0.smithi125.stdout:                        {interval} is a period (in seconds) to update the
2021-08-16T20:03:44.596 INFO:tasks.workunit.client.0.smithi125.stdout:                          histogram, 1 sec if omitted
2021-08-16T20:03:44.597 INFO:tasks.workunit.client.0.smithi125.stdout:                        {count} determines how many times to print,
2021-08-16T20:03:44.597 INFO:tasks.workunit.client.0.smithi125.stdout:                          infinite if omitted
...
2021-08-16T20:03:44.933 INFO:tasks.workunit.client.0.smithi125.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:2733: test_osd_daemon_histogram:  ceph daemonhistogram osd.all osd.op_r_latency_out_bytes_histogram batch 1 3
2021-08-16T20:03:44.933 INFO:tasks.workunit.client.0.smithi125.stderr:no valid command found; 10 closest matches:
2021-08-16T20:03:44.933 INFO:tasks.workunit.client.0.smithi125.stderr:pg stat
2021-08-16T20:03:44.933 INFO:tasks.workunit.client.0.smithi125.stderr:pg getmap
2021-08-16T20:03:44.934 INFO:tasks.workunit.client.0.smithi125.stderr:pg dump [<dumpcontents:all|summary|sum|delta|pools|osds|pgs|pgs_brief>...]
2021-08-16T20:03:44.934 INFO:tasks.workunit.client.0.smithi125.stderr:pg dump_json [<dumpcontents:all|summary|sum|pools|osds|pgs>...]
2021-08-16T20:03:44.934 INFO:tasks.workunit.client.0.smithi125.stderr:pg dump_pools_json
2021-08-16T20:03:44.934 INFO:tasks.workunit.client.0.smithi125.stderr:pg ls-by-pool <poolstr> [<states>...]
2021-08-16T20:03:44.935 INFO:tasks.workunit.client.0.smithi125.stderr:pg ls-by-primary <id|osd.id> [<pool:int>] [<states>...]
2021-08-16T20:03:44.935 INFO:tasks.workunit.client.0.smithi125.stderr:pg ls-by-osd <id|osd.id> [<pool:int>] [<states>...]
2021-08-16T20:03:44.935 INFO:tasks.workunit.client.0.smithi125.stderr:pg ls [<pool:int>] [<states>...]
2021-08-16T20:03:44.935 INFO:tasks.workunit.client.0.smithi125.stderr:pg dump_stuck [<stuckops:inactive|unclean|stale|undersized|degraded>...] [<threshold:int>]
2021-08-16T20:03:44.935 INFO:tasks.workunit.client.0.smithi125.stderr:Error EINVAL: invalid command
2021-08-16T20:03:44.936 INFO:tasks.workunit.client.0.smithi125.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:1: test_osd_daemon_histogram:  rm -fr /tmp/cephtool.zlP
2021-08-16T20:03:44.936 INFO:tasks.workunit:Stopping ['cephtool'] on client.0...
```

This is still failing http://pulpito.front.sepia.ceph.com/yuriw-2021-08-16_19:17:02-rados-wip-yuri7-testing-2021-08-16-0828-distro-basic-smithi/6341015/

Reply:
@neha-ojha - yeah, thanks for looking. I think I found and fixed the root cause, looks ready for another QA run.

Reply:
sure!